<div style="background:linear-gradient(135deg,#2e1065 0%,#6d28d9 55%,#8b5cf6 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#ddd6fe;font-weight:700;text-transform:uppercase">Chapter 42 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Joint Densities, Worked Answers ✅</div>
  <div style="font-size:15px;color:#ede9fe;max-width:720px;line-height:1.6">Full solutions to the five challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#c4b5fd">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
rng = np.random.default_rng(421)
joint = np.array([[0.10, 0.05, 0.05],
                  [0.05, 0.20, 0.10],
                  [0.05, 0.10, 0.30]])   # rows = X in {0,1,2}, cols = Y in {0,1,2}
print("joint sums to", joint.sum())

joint sums to 1.0


<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#7c3aed;letter-spacing:1px">CHALLENGE 1 · READ THE JOINT</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">A specific pair</div>
<div style="color:#4a5578;margin-top:6px">Using the 3x3 joint above (X, Y each in {0,1,2}), find P(X=2, Y=2) and P(X=Y).</div>
</div>

In [2]:
print(f"P(X=2, Y=2) = {joint[2,2]:.2f}")
print(f"P(X=Y) = {np.trace(joint):.2f}")

P(X=2, Y=2) = 0.30
P(X=Y) = 0.60


**Answer:** P(X=2, Y=2) is the bottom-right cell, **0.30**. P(X=Y) sums the diagonal: 0.10 + 0.20 + 0.30 = **0.60**.

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#6d28d9;letter-spacing:1px">CHALLENGE 2 · MARGINALS</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Each variable alone</div>
<div style="color:#4a5578;margin-top:6px">Find the marginal distributions of X and Y.</div>
</div>

In [3]:
pX = joint.sum(axis=1); pY = joint.sum(axis=0)
print(f"P(X): {pX.round(2)}")
print(f"P(Y): {pY.round(2)}")

P(X): [0.2  0.35 0.45]
P(Y): [0.2  0.35 0.45]


**Answer:** summing each row gives P(X) = [0.20, 0.35, 0.45]; summing each column gives P(Y) = [0.20, 0.35, 0.45]. Both marginals are valid distributions summing to 1.

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0d9488;letter-spacing:1px">CHALLENGE 3 · CONDITIONAL</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Y given X</div>
<div style="color:#4a5578;margin-top:6px">Find the conditional distribution P(Y | X=2).</div>
</div>

In [4]:
pX = joint.sum(axis=1)
cond = joint[2]/pX[2]
print(f"P(Y | X=2) = {cond.round(3)}")

P(Y | X=2) = [0.111 0.222 0.667]


**Answer:** take the X=2 row [0.05, 0.10, 0.30] and divide by P(X=2)=0.45, giving P(Y | X=2) &approx; [0.111, 0.222, 0.667]. Given X=2, Y is most likely 2 as well, the variables are positively associated.

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · INDEPENDENT?</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Test the factorization</div>
<div style="color:#4a5578;margin-top:6px">Are X and Y independent? Compare the joint to the product of the marginals.</div>
</div>

In [5]:
pX=joint.sum(axis=1); pY=joint.sum(axis=0)
indep = np.outer(pX,pY)
print(f"product of marginals[0,0] = {indep[0,0]:.3f}, actual joint[0,0] = {joint[0,0]:.3f}")
print(f"independent? {np.allclose(joint, indep)}")

product of marginals[0,0] = 0.040, actual joint[0,0] = 0.100
independent? False


**Answer:** the product of marginals gives 0.04 in the top-left cell, but the actual joint has 0.10, so the joint does **not** factor into the marginals. X and Y are **dependent** (they tend to move together, as the conditional in Challenge 3 showed).

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#5b21b6;letter-spacing:1px">CHALLENGE 5 · CORRELATION</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">From a simulated joint</div>
<div style="color:#4a5578;margin-top:6px">Simulate 5000 draws from a bivariate normal with correlation 0.6 and recover the correlation from the sample.</div>
</div>

In [6]:
cov=[[1,0.6],[0.6,1]]
xy=rng.multivariate_normal([0,0], cov, size=5000)
print(f"sample correlation = {np.corrcoef(xy.T)[0,1]:.3f}  (target 0.6)")
print(f"sample covariance matrix:\n{np.cov(xy.T).round(3)}")

sample correlation = 0.587  (target 0.6)
sample covariance matrix:
[[0.995 0.579]
 [0.579 0.98 ]]


**Answer:** the recovered correlation is close to the target 0.6, and the off-diagonal of the sample covariance matrix is near 0.6 as well. Correlation is the standardized covariance, a unitless measure of how two variables move together, and it is read straight off the joint distribution.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You read joint probabilities, recovered marginals and conditionals, tested independence by factorization, and measured correlation from a simulated joint. The joint distribution holds all the information; marginals and conditionals are just two ways of looking at it.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>